### ISSUE TITLE
LoadError on chef/resource/scm in Chef 16

### ISSUE THREAD

**Issue by jtimberman** [chef-cookbooks/deploy_resource#11]
### Cookbook version
1.0.2

### Chef-client version
16.0.257

### Platform Details
Ubuntu 16.04

### Scenario:
Cannot converge a cookbook using the `deploy_revision` resource after upgrading to Chef 16. I am using Test Kitchen to test my cookbook locally after experiencing this problem in a production system.

```
================================================================================
Recipe Compile Error in /opt/kitchen/cache/cookbooks/deploy_resource/libraries/resource_deploy.rb
================================================================================

LoadError
---------
cannot load such file -- chef/resource/scm

Cookbook Trace:
---------------
  /opt/kitchen/cache/cookbooks/deploy_resource/libraries/resource_deploy.rb:39:in `<top (required)>'

Relevant File Content:
----------------------
/opt/kitchen/cache/cookbooks/deploy_resource/libraries/resource_deploy.rb:

 32:  #   restart_command "touch tmp/restart.txt"
 33:  #   git_ssh_wrapper "wrap-ssh4git.sh"
 34:  #   scm_provider Chef::Provider::Git # is the default, for svn: Chef::Provider::Subversion
 35:  #   svn_username "whoami"
 36:  #   svn_password "supersecret"
 37:  # end
 38:
 39>> require "chef/resource/scm"
 40:
 41:  class Chef
 42:    class Resource
 43:
 44:      # Deploy: Deploy apps from a source control repository.
 45:      #
 46:      # Callbacks:
 47:      # Callbacks can be a block or a string. If given a block, the code
 48:      # is evaluated as an embedded recipe, and run at the specified

System Info:
------------
chef_version=16.0.257
platform=ubuntu
platform_version=16.04
ruby=ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]
program_name=/opt/chef/embedded/bin/chef-client
executable=/opt/chef/embedded/bin/chef-client
```

It appears that in Chef 16, `lib/chef/resource/scm.rb` was .... moved to `lib/chef/resource/scm/_scm.rb`?

**Comment by jtimberman** [comment]
Tl;dr, Don't use this cookbook with Chef 16+.

**Comment by phenggeler** [comment]
@jtimberman - sorry to piggyback on this ticket, but I am encountering this issue. Is there any documentation on what cookbook we should be using with Chef 16+? What should we use instead? Thanks!

**Comment by jtimberman** [comment]
I don't know. I don't work at Chef anymore.
